### DISCUSSION TITLE
Generated list of angles/axes of BCC variant pairs not unique for K-S OR

### DISCUSSION THREAD

**Discussion by AzdiarGazder** [mtex-toolbox/mtex#251] [Ask Anything] [answered]
Dear Developers,

Given below is a script written in Mtex 4.5.0 to generate a list of angle/axes shared between the 24 bcc variants for the K-S orientation relationship. The angle/axis notation is representative of the unique misorientation angle and axis shared by a pair of variants V1/Vx, where (V = variant, x varies from 1 to 24).

```matlab
%--- Start Mtex
startup_mtex;
%---

%--- Define the crystal symmetry of the various phases (Note: Phase ID should match the CTF file)
cs_fcc = crystalSymmetry('m-3m', [3.6599 3.6599 3.6599], 'mineral', 'fcc');
cs_bcc = crystalSymmetry('m-3m', [2.866 2.866 2.866], 'mineral', 'bcc');
%---

%--- Define sample symmetry
ss = specimenSymmetry('orthorhombic');
%---

%--- Define the orientation relationship ---
% The Kurdjumov-Sachs OR
fcc2bcc = orientation('map',...
  Miller(1,1,1,cs_fcc),Miller(0,1,1,cs_bcc),...
  Miller(1,0,-1,cs_fcc),Miller(1,1,-1,cs_bcc));

fcc2_BCCvariants = fcc2bcc * cs_fcc.properGroup * inv(fcc2bcc);
anglz_fcc2_BCCvariants = angle(fcc2_BCCvariants)./degree
axz_fcc2_BCCvariants = axis(fcc2_BCCvariants);
[axz_fcc2_BCCvariants.h axz_fcc2_BCCvariants.k axz_fcc2_BCCvariants.l]`
```

The script outputs the angles and axes of the 24 variant pairs of the K-S OR in the command window.
I noticed that the outputted list is not unique - i.e. - there are 2 sets of 2 variant pairs (highlighted in bold below) that return the same angle/axis.

For example:
In a unique K-S OR list, there are 3 variant pairs with a 60 deg misorientation as follows:
```matlab
60 deg/ [-1 -1 1]
60 deg/ [0 -1 -1]
60 deg/ [0 1 1]
```
In the output from the script, there are 3 variant pairs with a 60 deg misorientation as follows:
60 deg/ [-1.6547 1.6547  1.6547]; which can be re-written as 60 deg/ [-1 1 1]
**60 deg/ [-2.0266 0.00 2.0266]; which can be re-written as 60 deg/ [-1 0 1]
60 deg/ [-2.0266 0.00 2.0266]; which can be re-written as 60 deg/ [-1 0 1]**
So, the 2nd and 3rd of these variant pairs (in bold) have exactly the same angles and axes.

Similarly, in a unique K-S OR list, there are 4 variant pairs with a 50.5102 deg misorientation as follows:
50.5102 deg/ ~[-2 -1 1]
50.5102 deg/ ~[-1 1 -2]
50.5102 deg/ ~[-3 1 -4]
50.5102 deg / ~[-1 4 3]

In the output from the script, there are 4 variant pairs with a 50.5102 deg misorientation as follows:
50.5102 deg/ [-0.5338 1.7612 2.1971]; which can be approximately re-written as 50.5102 deg/ ~[-1 3 4]
50.5102 deg/ [-1.7612 0.5338 2.1971]; which can be approximately re-written as 50.5102 deg/ ~[-3 1 4]
**50.5102 deg/ [-1.4054 1.3254 2.1171]; which can be approximately re-written as 50.5102 deg/ ~[-1 1 2]
50.5102 deg/ [-1.3254 1.4054 2.1171]; which can be re-written as 50.5102 deg/ ~[-1 1 2]**
So, the 3rd and 4th of these variant pairs (in bold) have the same angle and approximately the same axes.

I do not know and am not sure if this is a bug in Mtex (so my apologies in advance if I have posted this issue in the wrong forum) but I would be very grateful for:
- any help to understand why the output is not returning 24 unique angles/axes and,
- edits to the script that resolves this issue.

Thank you in advance for any and all help.
Looking forward to hearing from you.

Regards,
Azdi

**Comment by ralfHielscher** [comment]
Hi Azdiar,

sorry for the late answer, but it took me some time to understand this issue.

I think the reason you get a different result as expected is that what you compute as ```fcc2_BCCvariants``` are **misorientations** from variant 1 to all other variants. And indeed misorientation 2 and 3 in your list are symmetrically equivalent. 

On the other hand, variants usually refers to **orientations**, i.e., assume variant 1 to have cube orientation (0°,0°,0°) then all other variants are computed by
```matlab
v = orientation.id(cs_bcc) * fcc2_BCCvariants
```
These variants are **not** symmetrically equivalent. Inspecting their rotational angles and axes I think you get close to what you expect.
```matlab
omega = angle(v)./degree;
[omega,id] = sort(omega);
ax = round(axis(v(id)),'maxHKL',4);
[omega, ax.h, ax.k, ax.l]
```
Ralf

**Comment by AzdiarGazder** [comment]
Hi Ralf,

Thank you ever so much for looking into this.
I understand that my error is in defining misorientations as opposed to each variant being an orientation.
I applied the modified lines of code to the Burgers OR and encountered a different problem as follows:

```matlab
%--- Start Mtex
startup_mtex;
%---

%--- Define the crystal symmetry of the various phases (Note: Phase ID should match the CTF file)
cs_hcp = crystalSymmetry('6/mmm', [2.545 2.545 4.14], 'X||a', 'Y||b*', 'Z||c', 'mineral', 'hcp','color',[0.5  0.5  0.5]);
cs_bcc = crystalSymmetry('m-3m', [2.866 2.866 2.866], 'mineral', 'bcc');
%---

%--- Define sample symmetry
ss = specimenSymmetry('orthorhombic');
%---

%--- Define the orientation relationship ---
% Note: the following command requires only one set of orthogonal planes and directions to describe the full OR
% The Burgers OR
hcp2bcc = orientation('map',...
  Miller(0,0,0,1,cs_hcp,'HKIL',ss),Miller(0,1,1,cs_bcc,'hkl',ss),...
  Miller(1,1,-2,0,cs_hcp,'UVTW',ss),Miller(1,1,-1,cs_bcc,'uvw',ss));

hcp2_BCCvariants = hcp2bcc * cs_hcp.properGroup * inv(hcp2bcc);

v = orientation.id(cs_bcc.properGroup) * hcp2_BCCvariants;

omega = angle(v)./degree;
[omega,id] = sort(omega);
ax = axis(v(id));
[omega', ax.h, ax.k, ax.l]`
```
The output is as follows:
```
    0.0000    1.0000         0         0
    0.0000    1.0000         0         0

   10.5288    0.0000   -0.7071   -0.7071
   10.5288    0.0000   -0.7071   -0.7071

   49.4712    0.0000    0.7071    0.7071
   49.4712    0.0000    0.7071    0.7071

   60.0000    0.0000    0.7071    0.7071
   60.0000    0.5774    0.5774   -0.5774 <---
   60.0000    0.0000    0.7071    0.7071
   60.0000   -0.5774   -0.5774    0.5774 <--
   60.0000    0.0000   -0.7071   -0.7071
   60.0000    0.0000   -0.7071   -0.7071
```
It is logical that for each misorientation angle, there are 2 listed angle/axis outputs that are the same. This can be easily reduced down to one unique angle/axis per misorientation angle.
So a reduced list of 6 unique variants can be produced for this orientation relationship.

In the output, for the 60 deg misorientation angle, there are 6 listed variants which need to be reduced down to 3 unique variants.
4 of the listed variants with 60 deg misorientation can be reduced to 2 unique variants and are not a problem:
   60.0000    0.0000    0.7071    0.7071 (rows 7 and 9)
   60.0000    0.0000   -0.7071   -0.7071 (rows 11 and 12)

However, rows 8 and 10 (in bold) should return the same angle and axes in order to reduce them to 1 unique variant. But their axes do not match and are re-listed below as follows:
   **60.0000    0.5774    0.5774   -0.5774**
   **60.0000   -0.5774   -0.5774    0.5774**

Is there something else that needs to be considered when dealing with hexagonal-cubic transformations and their variants that I have not accounted for?

Once again, thank you so much for all your help.

Cheers,
Azdi

**Comment by ralfHielscher** [comment]
The point is that for 60 degree rotations we are at the boundary of the fundamental region of the orientation space and hence the misorientation axis is well defined only up to sign.

The most simples way to reduce your variants to a unique list of variants is
```matlab
v = unique(v)
```
right after the definition of `v`.

Ralf.

**Comment by ks-or** [comment]
Dear Mr. Hielscher,

i want to calculate the angles between KS-variants. I am referring to this publication for the definition and numbering of the variants:

Morito, S., Huang, X., Furuhara, T., Maki, T., & Hansen, N. (2006). The morphology and crystallography of lath martensite in alloy steels. Acta Materialia, 54(19), 5323–5331. http://doi.org/10.1016/j.actamat.2006.07.009

The definition is done using


```
iks(1)  = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,0,1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(2)  = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,0,1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(3)  = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,1,-1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(4)  = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,1,-1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(5)  = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,-1,0,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(6)  = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,-1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(7)  = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,0,-1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(8)  = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,0,-1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(9)  = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,-1,0,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(10) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,-1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(11) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,1,1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(12) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,1,1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(13) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,-1,1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(14) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,-1,1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(15) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,0,-1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(16) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,0,-1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(17) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,1,0,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(18) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(19) = orientation('map',Miller(1,1,-1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,1,0,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(20) = orientation('map',Miller(1,1,-1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(-1,1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(21) = orientation('map',Miller(1,1,-1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,-1,-1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(22) = orientation('map',Miller(1,1,-1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(0,-1,-1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
iks(23) = orientation('map',Miller(1,1,-1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,0,1,bccCS,'uvw'),Miller(-1,-1,1,fccCS,'uvw'));
iks(24) = orientation('map',Miller(1,1,-1,bccCS,'hkl'),Miller(0,rx,ry,fccCS,'hkl'),Miller(1,0,1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
```

The paper mentioned above also gives the angles and axis of the variants with regard to variant 1. If I do the same using MTEX I use the following commands:

```
trans = orientation('Miller',[1 1 1],[-1 0 -1],fccCS)*iks;

for i = 1:length(trans)
    r(i) = inv(trans(i))*trans(1);
end

table((r.angle/degree)',Miller2Text(round(r.axis)))
```
This gives:

 ```
angle            axis     
    __________    ______________

             0     1     0     0
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
    1.7075e-06     0     0     1
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
    1.7075e-06     0     0     1
        10.529    -1     1     1
    1.7075e-06     0     0     1
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1
             0     1     0     0
        10.529    -1     1     1

```
This is not consistent with the angles given in the paper, which are:
```
	—	—
60.00	[0.5774–0.57740.5774]
60.00	[0.0000–0.7071-0.7071]
10.53	[0.00000.70710.7071]
60.00	[0.00000.70710.7071]
49.47	[0.0000–0.7071–0.7071]
49.47	[-0.5774–0.57740.5774]
10.53	[0.5774–0.57740.5774]
50.51	[−0.18620.76660.6145]
50.51	[−0.4904–0.46250.7387]
14.88	[0.3543–0.9329–0.0650]
57.21	[0.3568–0.71360.6029]
14.88	[0.93290.35430.0650]
50.51	[-0.73870.4625-0.4904]
57.21	[-0.2461-0.6278-0.7384]
20.61	[0.65890.65890.3628]
51.73	[-0.65890.3628-0.6589]
47.11	[-0.3022-0.6255-0.7193]
50.51	[-0.61450.1862-0.7666]
57.21	[-0.3568-0.6029-0.7136]
20.61	[0.95510.0000-0.2962]
47.11	[-0.71930.3022-0.6255]
57.21	[-0.7384-0.24610.6278]
21.06	[0.91210.41000.0000]
```

plotting the orientations from a arbitrary fcc-orientation gives only two single orientations in a pole figure

```
figure;plotPDF(orientation('Miller',[1 1 1],[-1 0 -1],fccCS)*iks,Miller(1,1,0,bccCS),'complete','all')
```

![ks1](https://user-images.githubusercontent.com/33747345/32949183-345461d2-cba2-11e7-8828-ac995bbafcf9.jpg)

- Is this a bug in the plotPDF-function?
- Will this also affect misorientation analysis (identifying variants in EBSD-Scans)?
- Is there a way to calculate the angles and axis between the variants according to the paper using MTEX?
- Is it possible to visualize misorientations ignoring symmetry in MTEX?
- Can MTEX distinguish between misorientation and disorientation?
- Can I plot pole figures representing ORs similar to Fig. 2 in the paper mentioned above?

The result is not suprising because

```
unique(orientation('Miller',[1 1 1],[-1 0 -1],fccCS)*iks)
 
ans = orientation (show methods, plot)
  size: 1 x 2
  crystal symmetry : Iron bcc (old) (m-3m)
  specimen symmetry: 1
 
  Bunge Euler angles in degree
     phi1     Phi    phi2    Inv.
  64.1169 64.6496 84.5071       0
  340.003 87.5211 31.0067       0

```
So I guess that the definition of the variants in the paper includes only two variants which are not symmetrically equivalent (but they also show a pole figure representing the variants and calculate angles between them)? Is there a problem in the paper or in MTEX?
Is there a way to define variants which are consistently grouped and numbered comparable to the literature?


If I use another definition from here: [http://digitool.library.mcgill.ca/webclient/StreamGate?folder_id=0&dvs=1510652225996~435](http://digitool.library.mcgill.ca/webclient/StreamGate?folder_id=0&dvs=1510652225996~435) (p. 31)

which gives:
```
yks(2) = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(-1,0,1,fccCS,'hkl'),Miller(-1,0,1,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'));
yks(15) = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(-1,1,0,fccCS,'hkl'),Miller(-1,1,0,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'));
yks(5) = orientation('map',Miller(-1,-1,1,bccCS,'hkl'),Miller(1,0,1,fccCS,'hkl'),Miller(1,0,1,bccCS,'uvw'),Miller(1,1,-1,fccCS,'uvw'));
yks(18) = orientation('map',Miller(-1,-1,1,bccCS,'hkl'),Miller(1,-1,0,fccCS,'hkl'),Miller(1,-1,0,bccCS,'uvw'),Miller(1,1,-1,fccCS,'uvw'));
yks(8) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(1,0,1,fccCS,'hkl'),Miller(1,0,1,bccCS,'uvw'),Miller(1,-1,-1,fccCS,'uvw'));
yks(21) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(1,1,0,fccCS,'hkl'),Miller(1,1,0,bccCS,'uvw'),Miller(1,-1,-1,fccCS,'uvw'));
yks(11) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(-1,0,1,fccCS,'hkl'),Miller(-1,0,1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
yks(24) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(-1,-1,0,fccCS,'hkl'),Miller(-1,-1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
yks(3) = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(1,-1,0,fccCS,'hkl'),Miller(1,-1,0,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'));
yks(13) = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,-1,1,fccCS,'hkl'),Miller(0,-1,1,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'));
yks(6) = orientation('map',Miller(-1,-1,1,bccCS,'hkl'),Miller(-1,1,0,fccCS,'hkl'),Miller(-1,1,0,bccCS,'uvw'),Miller(1,1,-1,fccCS,'uvw'));
yks(16) = orientation('map',Miller(-1,-1,1,bccCS,'hkl'),Miller(0,1,1,fccCS,'hkl'),Miller(0,1,1,bccCS,'uvw'),Miller(1,1,-1,fccCS,'uvw'));
yks(9) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(-1,-1,0,fccCS,'hkl'),Miller(-1,-1,0,bccCS,'uvw'),Miller(1,-1,-1,fccCS,'uvw'));
yks(19) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,-1,1,fccCS,'hkl'),Miller(0,-1,1,bccCS,'uvw'),Miller(1,-1,-1,fccCS,'uvw'));
yks(12) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(1,1,0,fccCS,'hkl'),Miller(1,1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
yks(22) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,1,1,fccCS,'hkl'),Miller(0,1,1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
yks(1) = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(0,1,-1,fccCS,'hkl'),Miller(0,1,-1,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'));
yks(14) = orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(1,0,-1,fccCS,'hkl'),Miller(1,0,-1,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'));
yks(4) = orientation('map',Miller(-1,-1,1,bccCS,'hkl'),Miller(0,-1,-1,fccCS,'hkl'),Miller(0,-1,-1,bccCS,'uvw'),Miller(1,1,-1,fccCS,'uvw'));
yks(17) = orientation('map',Miller(-1,-1,1,bccCS,'hkl'),Miller(-1,0,-1,fccCS,'hkl'),Miller(-1,0,-1,bccCS,'uvw'),Miller(1,1,-1,fccCS,'uvw'));
yks(7) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(0,1,-1,fccCS,'hkl'),Miller(0,1,-1,bccCS,'uvw'),Miller(1,-1,-1,fccCS,'uvw'));
yks(20) = orientation('map',Miller(-1,1,1,bccCS,'hkl'),Miller(-1,0,-1,fccCS,'hkl'),Miller(-1,0,-1,bccCS,'uvw'),Miller(1,-1,-1,fccCS,'uvw'));
yks(10) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(0,-1,-1,fccCS,'hkl'),Miller(1,-1,0,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
yks(23) = orientation('map',Miller(1,-1,1,bccCS,'hkl'),Miller(1,0,-1,fccCS,'hkl'),Miller(1,0,-1,bccCS,'uvw'),Miller(-1,1,-1,fccCS,'uvw'));
```
This gives the expected figure for

```
figure;plotPDF(orientation('Miller',[1 1 1],[-1 0 -1],fccCS)*yks,Miller(1,1,0,bccCS),'complete','all')
```

![ks2](https://user-images.githubusercontent.com/33747345/32949244-74c7aa30-cba2-11e7-91c4-69c5c1f5311e.jpg)

and the angles/axis are:

```
ans = 

     angle          axis   
    ______    ______________

         0     1     0     0
        60    -1     0     1
        60    -1     0     1
    20.605    -9     5     9
    47.113    -3     6     7
     50.51    -7     7    11
    10.529    -1     1     1
     50.51    -7     7    11
    57.212    -5     3     6
    21.058    -4     0     9
    57.212    -3     5     6
    47.113    -6     3     7
        60    -1     1     1
    49.471    -1     0     1
    10.529    -1     0     1
    57.212    -2     5     6
    51.729    -9     5     9
    14.879    -1     4    11
    49.471    -1     1     1
     50.51    -7     2     9
    14.879    -4     1    11
    57.212    -5     2     6
     50.51    -2     7     9
    20.605    -3     0    10

```

Which are again not consistent with the angles given in literature (p.59) which are:

```
Var No.		Literature	MTEX
1		0		0
2		60		60
3		60		60
4		21.06		20.605
5		47.11		47.113
6		57.21		50.51
7		10.53		10.529
8		57.21		50.51
9		50.51		57.212
10		20.61		21.058
11		50.51		57.212
12		47.11		47.113
13		60		60
14		10.53		49.471
15		49.47		10.529
16		57.21		57.212
17		20.61		51.729
18		50.51		14.879
19		49.47		49.471
20		14.88		50.51
21		50.51		14.879
22		57.21		57.212
23		14.88		50.51
24		51.73		20.605

```
Why are these angles not consistent?

The safest way to define KS-Varinats in MTEX seems to be something like:

```
ori = orientation('Miller',[0 0 1],[1 0 0],fccCS);
mtexks = inv(ori)*unique(ori*symmetrise(orientation('map',Miller(1,1,1,bccCS,'hkl'),Miller(-1,0,1,fccCS,'hkl'),Miller(-1,0,1,bccCS,'uvw'),Miller(-1,-1,-1,fccCS,'uvw'))));

```
which is nice and easy but how can i relate these variants to the definitions which are very frequently used in literature?
(e.g. Morito, S., Huang, X., Furuhara, T., Maki, T., & Hansen, N. (2006). The morphology and crystallography of lath martensite in alloy steels. Acta Materialia, 54(19), 5323–5331. http://doi.org/10.1016/j.actamat.2006.07.009)

Thanks in advance for your help!

Regards,
Michael

**Comment by ralfHielscher** [comment, marked as answer]
Hi Michel,

this is a long question. Some remarks not covering all:

1. You did not specify what rx and ry is. I assumed rx = ry = 1
2. to get the misorientations of variant 1 to all other variants do do not need to compute orientations. The following code gives the same output as your code
``` matlab
r = inv(iks(1)) .* iks
```
3. You get the output of the paper if you do
``` matlab
r = iks(1) .* inv(iks)
```
Although the result seems to be correct, it is not the correct way of computing the angles between the variants. What you did in 2) was the correct way. This indicates that the variants iks have not been defined correctly.

4. the plotPDF command indeed shows only two different orientations. You have seen this also with
``` matlab
unique(trans)
```
and your conclusion is correct the definition of the variants is not correct.

5. Is it possible to visualize misorientations ignoring symmetry in MTEX? - In the version I'm going to publish today there is an option ```noSymmetry```
``` matlab
plotPDF(mori,Miller(1,1,1,mori.CS),'noSymmetry')
```
6. Can MTEX distinguish between misorientation and disorientation? - sure

7. Can I plot pole figures representing ORs similar to Fig. 2 in the paper mentioned above? - Yes with the orientationrelationship properly defined.

8. Your second definition seems to be correct, only the angles are a bit permuted. 

9. In MTEX the new version provides
``` matlab
KS = orientation.KurdjumovSachs(cs_fcc,cs_bcc)
child_ori = parent_ori * inv(KS.variants)

angle(KS.variants * inv(KS))./degree
```
which gives you the angles you want - but in the wrong order.

10. I do understand your frustration. Maybe an option is to define a consistent order of the variants, e.g. according to the misorientation angle, and use ```sort``` to order the variants accordingly.

Ralf.

**Comment by ralfHielscher** [comment]
Hi Michael,

no further comment here?

Ralf.
